sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -508,7 +508,7 @@ pi_result _pi_context::finalize() {
   // For example, zeEventPool could be still alive.
   std::lock_guard<std::mutex> NumEventsLiveInEventPoolGuard(
       NumEventsLiveInEventPoolMutex, std::adopt_lock);
-  if (ZeEventPool && NumEventsLiveInEventPool[ZeEventPool])
+  if (ZeEventPool)
```

Comment:
Don't you need to destroy the events in the event poll too?

Reply:
I believe events are properly released.
I haven't seen any memory leak for events.
Further, I don't know how to get a list of live zeEvent objects from the zeEventPool.
ZE-APIs do not provide iterations of zeEvents in the pool or checking if a certain event is live.

Reply:
Each time an event is destroyed (after final release) it's removed from the `NumEventsLiveInEventPool` see in `_pi_context::decrementAliveEventsInPool`. So if you needed to remove the this check here should mean there were still events that were not released/destroyed. We should do something about them.